utils.py

```diff
@@ -22,21 +22,48 @@ def failure_response(message, code=404):
     return json.dumps({"success": False, "error": message}), code
 
 
-def upload_image_helper(image_data, bucket_name):
+def upload_image_helper(bucket_name,image_data=None, file_data=None):
+    """
+    Function that uploads image data to s3 based on the bucket name and data.
+    
+    Parameters
+    ----
+    bucket_name : str
+    The name of the bucket to place the image in
+    image_data : str | None
+    The raw base 64 passed in from the HTML post request, potentially None if
+    the client is using form-data
+    file_data: FileStore | None
+    The file from the form-data of the user's HTML request, or None if it was
+    passed in the request body instead.
+    
+    Returns
+    ----
+    The image URL associated to the image that was uploaded to s3
+    """
     if bucket_name not in BUCKET_NAMES:
         return None
-    mime_type = guess_type(image_data)[0]
-    ext = guess_extension(guess_type(image_data)[0])[1:]
-    if re.fullmatch(ALLOWED_MIME_TYPES_REGEX, mime_type) is None:
-        raise Exception(f"Extension {ext} not supported!")
-
+    # Guess the file type, depending on whether we have image data or file data
+    if image_data:
+        mime_type = guess_type(image_data)[0]
+        ext = guess_extension(guess_type(image_data)[0])[1:]
+        if re.fullmatch(ALLOWED_MIME_TYPES_REGEX, mime_type) is None:
+            raise Exception(f"Extension {ext} not supported!")
+    else:
+        mime_type = guess_type(file_data.filename)[0]
+        ext = guess_extension(mime_type)[1:]
+        if re.fullmatch(ALLOWED_MIME_TYPES_REGEX, mime_type) is None:
+            raise Exception(f"Extension {ext} not supported!")
```

Comment:
Note that this code is essentially duplicated since it has essentially the same structure.